RUNBOOK.md

```diff
@@ -0,0 +1,153 @@
+
+### Technical overview
+
+Internal API based on Springboot, using Kotlin as the main language. Deployed in kubernetes using the configuration found in [helm_deploy](helm_deploy).
+
+### Service Level Agreements (SLAs)
+
+Office hours (Mon-Fri, 09:00-17:00), best efforts.
+
+### Service owner
+
+The `ppud-replacement` team develops and runs this service.
+
+Contact the [#ppud-replacement](https://mojdt.slack.com/archives/C020S8C0K9U) and [#ppud-replacement-devs](https://mojdt.slack.com/archives/C0223AGGQU8) channels on slack.
+
+### Contributing applications, daemons, services, middleware
+
+- Springboot application based on [hmpps-template-kotlin](https://github.com/ministryofjustice/hmpps-template-kotlin).
+- PostgreSQL database for persistence - configured via [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments).
+- AWS S3 for document storage - again, configured via [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments).
+- [gotenberg](https://gotenberg.dev/) for PDF rendering - included via a helm sub-chart.
+- [ClamAV](https://www.clamav.net/) for AV scanning of uploaded files - included via a helm chart.
+- [CircleCI](https://circleci.com/) for CI/CD.
+
+## System characteristics
+
+### Hours of operation
+
+_TBC_
+
+### Infrastructure design
+
+The application runs on the [HMPPS Cloud Platform](https://user-guide.cloud-platform.service.justice.gov.uk/) within the `ppud-replacement-<tier>` namespaces (where `tier` is `dev`, `preprod` or `prod`).
+
+The main application runs as a deployment named `manage-recalls-api`, with the following support deployments (all deployed as part of the same helm/kubernetes configuration):
+
+- `manage-recalls-api-gotenberg` - for PDF rendering.
+- `manage-recalls-api-clamav` - for AV scanning.
+
+The API is made avialable externally from the cluster via an Ingress.
+
+See the `values-<tier>.yaml` files in the [helm_deploy](helm_deploy) directory for the current configuration of each tier.
+
+### Throttling and partial shutdown
+
+If there is an issue with the service where it is causing load on downstream services and it needs to be shutdown quickly the following command will reduce the number of pod replicas to zero:
+
+```
+kubectl -n ppud-development-<tier> scale deployment manage-recalls-api --replicas=0
+```
+
+We do not currently have a strategy in place to throttle requests.
+
+### Environmental differences
+
+Infrastructure wise, all three tiers are identical, but `prod` has the following differences:
+
+- It will have more pod replicas of the main application deployment.
+- As this is live data, you **must** be SC cleared to work on it or access the data held within it.
+
+### Tools
+
+- [port-forward-db.sh](scripts/port-forward-db.sh) - allows you to connect to the PostreSQL database from your local machine (as external access is blocked).
```

Comment:
do we need to mention the environment variable? eg. -e preprod

Reply:
That information is in the script - no need to repeat it here otherwise it's likely to get out of date. :)